drivers/dma/dma_mcux_lpc.c

```diff
@@ -565,7 +576,7 @@ static const struct dma_mcux_lpc_config dma_##n##_config = {	\
 			    &dma_mcux_lpc_init,				\
 			    NULL,					\
 			    &dma_data_##n, &dma_##n##_config,\
-			    POST_KERNEL, CONFIG_DMA_INIT_PRIORITY,	\
+			    PRE_KERNEL_1, CONFIG_DMA_INIT_PRIORITY,	\
```

Comment:
We cannot initialize this driver in the PRE_KERNEL phase, as it calls `k_malloc` in the init function: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/dma/dma_mcux_lpc.c#L486

Reply:
good catch, I removed the k_malloc calls and replaced them with static allocations in the driver initialization macro